### ISSUE TITLE
Long path error in symbols generation of dotnet store

### ISSUE THREAD

**Issue by JunTaoLuo** [dotnet/sdk#1302]
From the offline discussion there seem to be a race condition during the copy of crossgen pdb to the staging dir. This produces errors like:
```
    Error generating PDB for 'C:\gh\MetaPackages2\src\Microsoft.AspNetCore.RuntimeStore\bin\work\Build.RuntimeStore.References_2.0.0--\runtimopt\Microsoft.AspNetCore.Authentication.Abstractions.dll': Unspecified error (Exception from HRESULT: 0x80004005 (E_FAIL))
EXEC : error : compilation failed for "C:\gh\MetaPackages2\src\Microsoft.AspNetCore.RuntimeStore\bin\work\Build.RuntimeStore.References_2.0.0--\runtimopt\Microsoft.AspNetCore.Authentication.Abstractions.dll" (0x80004005) [C:\gh\MetaPackages2\src\Microsoft.AspNetCore.RuntimeStore\bin\work\Restore.csproj] [C:\gh\MetaPackages2\.build\KoreBuild.proj]
C:\Users\johluo\.dotnet\x64\sdk\2.0.0-preview2-006215\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.CrossGen.targets(162,5): error MSB3073: The command "C:\Users\johluo\.nuget\packages\runtime.win-x64.microsoft.netcore.app\2.0.0-preview2-25401-01\tools\crossgen.exe -readytorun -platform_assemblies_paths C:\gh\MetaPackages2\src\Microsoft.AspNetCore.RuntimeStore\bin\work\Build.RuntimeStore.References_2.0.0--\runtimeref;C:\gh\MetaPackages2\src\Microsoft.AspNetCore.RuntimeStore\bin\work\Optimize\netcoreapp -CreatePDB C:\gh\MetaPackages2\src\Microsoft.AspNetCore.RuntimeStore\bin\work\Build.RuntimeStore.References_2.0.0--\runtimesymbols\microsoft.aspnetcore.authentication.abstractions\2.0.0-preview3-25366\lib\netstandard2.0 C:\gh\MetaPackages2\src\Microsoft.AspNetCore.RuntimeStore\bin\work\Build.RuntimeStore.References_2.0.0--\runtimopt\Microsoft.AspNetCore.Authentication.Abstractions.dll" exited with code -2147467259. [C:\gh\MetaPackages2\src\Microsoft.AspNetCore.RuntimeStore\bin\work\Restore.csproj] [C:\gh\MetaPackages2\.build\KoreBuild.proj]
```

We found this during our metapackage build https://github.com/aspnet/metapackages. Steps to repro:
1. Set environment variable PACKAGE_CACHE_PLATFORM=x64
2. Set environment variable BuildNumber=final
3. run ./build.cmd /t:BuildPackageCache

This is blocking our cross-gen'ed symbols generation for preview2.

cc @ramarag @eerhardt @bleroy @Eilon

**Comment by eerhardt** [comment]
@JunTaoLuo - this isn't a race condition problem, but instead it is a long path problem.

I changed the command in your repo from:

`dotnet store -m src\Microsoft.AspNetCore.RuntimeStore\Microsoft.AspNetCore.RuntimeStore.csproj -f netcoreapp2.0 -r win7-x64 -o F:\git\metapackages\src\Microsoft.AspNetCore.RuntimeStore\bin\packageCache --framework-version 2.0.0-* -w F:\git\metapackages\src\Microsoft.AspNetCore.RuntimeStore\bin\work`

to instead specify:

`-w C:\temp\store`

And the command executed successfully.

I've logged https://github.com/dotnet/coreclr/issues/12112 to make this the underlying error experience better when we run into a long path problem in this area.  At least in that case we won't get an unintelligible `Unspecified error (Exception from HRESULT: 0x80004005 (E_FAIL))` error message.

Leaving this bug open to see if there is any possibility of shortening our working directory paths.  But moving it out of `2.0` since this shouldn't be blocking ASP.NET anymore.  They can change their command to use a shorter working directory root path.
